core/deployment/src/main/java/io/quarkus/deployment/builditem/QuarkusBuildCloseablesBuildItem.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.deployment.builditem;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class QuarkusBuildCloseablesBuildItem extends SimpleBuildItem implements Closeable {
+
+    private static final Logger log = Logger.getLogger(QuarkusBuildCloseablesBuildItem.class);
+
+    private final List<Closeable> closeables = new ArrayList<>();
```

Comment:
This needs to be thread safe.

Normally we don't want mutable build items but for this use case this makes sense, as we only want these to be produced as side effects.